appserver/payara-appserver-modules/microprofile/healthcheck/src/main/java/fish/payara/microprofile/healthcheck/admin/SetMPHealthCheckConfiguration.java

```diff
@@ -133,6 +157,46 @@ public void execute(AdminCommandContext context) {
         } catch (TransactionFailure ex) {
             actionReport.failure(LOGGER, "Failed to update HealthCheck configuration", ex);
         }
+
+        // If everything has passed, scrap the subaction reports as we don't want to print them out
+        if (!actionReport.hasFailures() || !actionReport.hasWarnings()) {
+            actionReport.getSubActionsReport().clear();
+        }
+    }
+
+    private boolean defaultMicroprofileUserExists(ActionReport subActionReport, Subject subject) {
+        boolean exists = false;
+
+        CommandRunner.CommandInvocation invocation = commandRunner.getCommandInvocation("list-file-users", subActionReport, subject,
+                false);
+
+        ParameterMap parameters = new ParameterMap();
+        parameters.add("authrealmname", "file");
+
+        invocation.parameters(parameters).execute();
+
+        for (ActionReport.MessagePart message : subActionReport.getTopMessagePart().getChildren()) {
+            if (message.getMessage().equals(DEFAULT_USER_NAME)) {
+                exists = true;
+                break;
+            }
+        }
+
+        return exists;
+    }
+
+    private void createDefaultMicroprofileUser(ActionReport subActionReport, Subject subject) {
+        CommandRunner.CommandInvocation invocation = commandRunner.getCommandInvocation("create-file-user", subActionReport, subject,
+                false);
+
+        ParameterMap parameters = new ParameterMap();
+        parameters.add("groups", DEFAULT_GROUP_NAME);
+        parameters.add("userpassword", "mp");
```

Comment:
Don't know what the "userpassword" is but is sounds like a password of some kind - is it meant to appear in code (safe to be known by everyone)?

Reply:
When `securityenabled` is set to true, a username and password are required to access MicroProfile endpoints. By default, the username & password is set to `mp` and defined in the `file` realm.

Reply:
I understand. I just wonder how good a security is when everyone can read the password in our public sources. So do we expect the default to practically never be used? Wouldn't it be more secure to force configuration of a password in case security is enabled?

Reply:
> Do we expect the default to practically never be used?

Yes

> Wouldn't it be more secure to force configuration of a password in case security is enabled?

I agree that if default user not exist and security is enabled then prompting for default username and password would be good. But prompting for password functionality and `com.sun.enterprise.admin.cli.ProgramOptions` injected instance is available for commands which inherits `com.sun.enterprise.admin.cli.CLICommand` class.